### DISCUSSION TITLE
Potential Enhancements

### DISCUSSION THREAD

**Discussion by danielrherber** [danielrherber/dt-qp-project#4] [Ideas]
Proposed internal/external variable naming update
 - Objective function (F -> J)
 - States (Y -> X)
 - Number of states (ny -> nx)
 - Optimization variables (X -> Z)
 - Number of optimization variables (nx -> nz)
 - Additional linear equality constraints (setup.Y -> setup.linear_eq) (H is used in too many places)
 - Additional linear inequality constraints (setup.Z -> setup.linear_in) (G is used to several places and want consistency with equality constraints)
 - Parameter matrix in dynamics (setup.G -> setup.Bp)
 - Disturbances (setup.d -> setup.v); also potential add setup.Bv
 - Parameter structure (p -> auxdata)

Project rename:
 - DTQP-NL to indicate the capability to solve nonlinear dynamic optimization problems
 - Only update project files and not GitHub link?

**Comment by yonghoonlee** [comment]
I think changing variable names may clarify a lot of potential confusions.

**Reply by danielrherber** [reply]
Thanks for the feedback @yonghoonlee. With commit 48c76c2 added today focusing mostly on the potentially nonlinear elements, some of the external setup naming structure is underway. Any suggestions in this area are appreciated.

**Reply by danielrherber** [reply]
@yonghoonlee I finally have some updates on this change. Added a new branch with the draft DTQP input struct:
https://github.com/danielrherber/dt-qp-project/compare/master...new-input-struct

The goal for now is total backward compatibility, so it is more natural to define the problems, but the source code remains unchanged as the new format is converted back into the old format. See the following for a good overview of the new structure compared to the old one:
https://github.com/danielrherber/dt-qp-project/commit/0dbc4020eca6763469ce9fb51868e79457f164ad#diff-9ee0dcc6315a094790efe69780ff052ac760c97abe8e2e403a38b8f168b97623

A linear-quadratic example in this new format (also now using GitHub native LaTeX in the README):
https://github.com/danielrherber/dt-qp-project/tree/new-input-struct/examples/.new/linear-quadratic/anderson-moore-64

A nonlin example in this new format:
https://github.com/danielrherber/dt-qp-project/tree/new-input-struct/examples/.new/nonlin/alp-rider

If you have any feedback, please let me know.

**Reply by yonghoonlee** [reply]
Awesome! I just skimmed through the diff link. I will take a look more detail when I get some time. Thank you for all these improvements.

**Comment by yonghoonlee** [comment]
Project rename may bring potential fragmentations of versions. Maybe keeping one name and one integrated framework, but provide submodules for different class of usage might be my preference. But it is your call.

**Reply by yonghoonlee** [reply]
In addition, I cited the dtqp with github link. I think keeping the same link may help exposure of this code to other researchers.
